pettingzoo/utils/env.py

```diff
@@ -180,11 +180,14 @@ def agent_iter(self, max_iter: int = 2**63) -> AECIterable:
         """
         return AECIterable(self, max_iter)
 
-    def last(self, observe: bool = True) -> Tuple[ObsType, float, bool, Dict[str, Any]]:
+    def last(
+        self, observe: bool = True
+    ) -> Tuple[Optional[ObsType], float, bool, Dict[str, Any]]:
         """
         Returns observation, cumulative reward, done, info   for the current agent (specified by self.agent_selection)
         """
         agent = self.agent_selection
+        assert agent
```

Comment:
Is this just checking that it's not None? Is it a legitimate concern?

Reply:
Yes, it removes a typing error as it makes it clear that `None` is not allowed. Do you think `None` should be allowed here?
Just for reference, using assertions is part of the tricks to help the linter choosing the right types, see for example here: https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#built-in-types=

Reply:
TIL